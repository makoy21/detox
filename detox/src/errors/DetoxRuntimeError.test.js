const _ = require('lodash');
const DetoxRuntimeError = require('./DetoxRuntimeError');

describe(DetoxRuntimeError, () => {
  it('should format its fields to a single message if needed', () => {
    expect(_.mapValues(varietiesOfInstantiation(), 'message')).toMatchSnapshot();
  });

  function varietiesOfInstantiation() {
    return {
      'no args': new DetoxRuntimeError(),
      'empty object': new DetoxRuntimeError({}),
      'only message': new DetoxRuntimeError({ message: 'message' }),
      'message with hint': new DetoxRuntimeError({ message: 'message', hint: 'hint' }),
      'message with debug info': new DetoxRuntimeError({ message: 'message', debugInfo: 'debugInfo' }),
      'message with hint and debug info': new DetoxRuntimeError({ message: 'message', hint: 'hint', debugInfo: 'debugInfo' }),
      'toString': { message: new DetoxRuntimeError({ message: 'message', hint: 'hint', debugInfo: 'debugInfo' }).toString() },
    };
  }
});
