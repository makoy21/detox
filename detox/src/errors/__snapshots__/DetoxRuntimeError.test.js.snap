// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetoxRuntimeError should format its fields to a single message if needed 1`] = `
Object {
  "empty object": "DetoxRuntimeError: ",
  "message with debug info": "DetoxRuntimeError: message

debugInfo",
  "message with hint": "DetoxRuntimeError: message

HINT: hint",
  "message with hint and debug info": "DetoxRuntimeError: message

HINT: hint

debugInfo",
  "no args": "DetoxRuntimeError: ",
  "only message": "DetoxRuntimeError: message",
  "toString": "DetoxRuntimeError: message

HINT: hint

debugInfo",
}
`;
